src/dali/plugin/input/csv/nexo.py

```diff
@@ -0,0 +1,150 @@
+
+
+# types of transactions
+_INTEREST = "Interest"
+_LOCKING_TERM_DEPOSIT = "LockingTermDeposit"
+_UNLOCKING_TERM_DEPOSIT = "UnlockingTermDeposit"
+_FIXED_TERM_INTEREST = "FixedTermInterest"
+_DEPOSIT = "Deposit"
+
+class InputPlugin(AbstractInputPlugin):
+
+    __NEXO: str = "Nexo"
+
+    __TRANSACTION_ID_INDEX = 0
+    __TRANSACTION_TYPE_INDEX = 1
+    __CURRENCY_INDEX: int = 2
+    __AMOUNT_INDEX: int = 3
+
+    __TIMESTAMP_INDEX: int = 7
+
+    __DELIMITER = ","
+
+    def __init__(
+        self,
+        account_holder: str,
+        transaction_csv_file: str,
+        native_fiat: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder=account_holder, native_fiat=native_fiat)
+        self.__transaction_csv_file: str = transaction_csv_file
+        self.__logger: logging.Logger = create_logger(f"{self.__NEXO}/{self.account_holder}")
+
+    def load(self) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        last_withdrawal_fee: Optional[RP2Decimal] = None
+        with open(self.__transaction_csv_file, encoding="utf-8") as transaction_csv_file:
+            # read CSV with header and skip first row
+            lines = reader(transaction_csv_file)
+
+            # Skip header line
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                transaction_id: str = line[self.__TRANSACTION_ID_INDEX].strip()
+                transaction_type: str = line[self.__TRANSACTION_TYPE_INDEX].strip()
+                currency: str = line[self.__CURRENCY_INDEX].strip()
+                amount = line[self.__AMOUNT_INDEX].strip()
+                timestamp_with_timezone = f"{line[self.__TIMESTAMP_INDEX].strip()} -00:00"
+
+                if transaction_type in [_INTEREST, _FIXED_TERM_INTEREST]:
+                    result.append(
+                        InTransaction(
+                            plugin=self.__NEXO,
+                            unique_id=transaction_id,
+                            raw_data=raw_data,
+                            exchange=self.__NEXO,
+                            holder=self.account_holder,
+                            timestamp=timestamp_with_timezone,
+                            asset=currency,
+
+                            transaction_type=Keyword.INTEREST.name,
+                            # nexo does give us the spot price, but it's often 0 if subcent
+                            spot_price=Keyword.UNKNOWN.value,
+                            crypto_in=amount,
+                            fiat_fee="0",
+                        )
+                    )
+                elif transaction_type in [_LOCKING_TERM_DEPOSIT, _UNLOCKING_TERM_DEPOSIT]:
+                    # I don't think we need to record locking/unlocking deposits for term interest
```

Comment:
What are locking / unlocking deposits?

Reply:
I'll write an inline comment on this, but this is a Nexo thing: they "lock" your crypto in a "fixed term" deposit which earns higher interest.

i.e. these transactions just indicate that you cannot withdraw these funds while these are locked. So they effect your available balance.